src/app/auth/components/registration/registration.component.html

```diff
@@ -0,0 +1,182 @@
+<form [formGroup]="registrationForm" class="regestration-form">
+  <h2>Registration</h2>
+  <!-- Email Field -->
+
+  <mat-form-field>
+    <mat-label>Enter your email</mat-label>
+    <input matInput placeholder="example@email.com" formControlName="email" />
+    @if (registrationForm.get("email")?.hasError("required")) {
+      <mat-error>You must enter an email</mat-error>
+    }
+    @if (registrationForm.get("email")?.hasError("email")) {
+      <mat-error>Provide valid email</mat-error>
+    }
+  </mat-form-field>
+
+  <!-- Password Fields -->
+
+  <mat-form-field>
+    <mat-label>Enter your password</mat-label>
+    <input
+      matInput
+      formControlName="password"
+      [type]="hidepassword ? 'password' : 'text'"
+    />
+    <button
+      mat-icon-button
+      matSuffix
+      (click)="hidepassword = !hidepassword"
+      [attr.aria-label]="'Hide password'"
+      [attr.aria-pressed]="hidepassword"
+    >
+      <mat-icon>{{ hidepassword ? "visibility_off" : "visibility" }}</mat-icon>
+    </button>
+    @if (registrationForm.get("password")?.hasError("noLowerCase")) {
```

Comment:
You have too many if-s here, to simplify this part you can make errors array as a constant.  PASSWORD_ERRORS = [
    {code: "noLowerCase", message: "Password must include at least one lowercase letter"},
    {code: "noUpperCase", message: "Password must include at least one uppercase letter"},
    {code: "noDigit", message: "Password must include at least one digit"},
    {code: "noSimbol", message: "Password must include at least one special character (!@#?)"},
    {code: "tooShort", message: "Password must be at least 8 characters long"}
  ];
then in the For="let error of PASSWORD_ERRORS" and inside for use if If="registrationForm.get('password')?.hasError(error.code)" to check the code. After that in the if you can print error.message.
Make sure to have contacts in the separate file